drivers/gpio/gpio_stm32.c

```diff
@@ -308,15 +325,16 @@ static int gpio_stm32_config(struct device *dev, int access_op,
 			     u32_t pin, int flags)
 {
 	const struct gpio_stm32_config *cfg = dev->config->config_info;
+	int err = 0;
 	int pincfg;
 	int map_res;
 
 	if (access_op != GPIO_ACCESS_BY_PIN) {
 		return -ENOTSUP;
 	}
 
-	if ((flags & GPIO_POL_MASK) == GPIO_POL_INV) {
-		/* hardware cannot invert signal */
+	if (((flags & GPIO_INPUT) != 0) && ((flags & GPIO_OUTPUT) != 0)) {
+		/* Simultaneous pin in/out mode is not supported */
```

Comment:
Is disconnected (neither input nor output) supported?

Reply:
On some series (all except STM32F1), We can have alternate function or analog.
Analog is described as follows:
```
• The output buffer is disabled
• The Schmitt trigger input is deactivated, providing zero consumption for every analog
value of the I/O pin. The output of the Schmitt trigger is forced to a constant value (0).
• The weak pull-up and pull-down resistors are disabled
• Read access to the input data register gets the value “0”
```
So, I guess this matches disconnected.

Reply:
Looking at the STM32L4x5/6 manual section 8.3.10 I agree with @mnkp: it appears that simultaneous in/out is supported whenever configured for output, and output-only (input disconnected) is not supported.

I think the helpful thing here would be to basically ignore `GPIO_INPUT`, so `GPIO_DISCONNECTED` and `GPIO_INPUT` are equivalent, and `GPIO_OUTPUT` and `GPIO_OUTPUT | GPIO_INPUT` are equivalent.

Reply:
@pabigot, @mnkp, so from this point of view, seems that this is the case indeed for all series. Thanks for the hint.
Though, input allows push-pull, which is not the case in analog. So I don't think we can have `GPIO_DISCONNECTED` and `GPIO_INPUT` put equivalent.